Started 9.1

nmamro opened the pull request:
- List functions
- General HashTable functions
- Reading from file and some HT functions left

nmamro committed:
Started 9.1

- List functions
- General HashTable functions
- Reading from file and some HT functions left

nmamro committed:
Small changes

nmamro committed:
Finished

nmamro committed:
Deleted "govnokod"

nmamro committed:
Vector styleguide; Signed counter -> unsigned

yurii-litvinov commented on Lab9/Task1/Homework/HashTable.cpp:
```diff
@@ -0,0 +1,263 @@
+	}
+
+	return table;
+}
```
Это вполне можно было бы сделать и где-то ещё, поскольку знания о внутренней структуре хеш-таблицы это не требует. Есть принцип минимизации интерфейсов :)

yurii-litvinov commented on Lab9/Task1/Homework/HashTable.cpp:
```diff
@@ -0,0 +1,263 @@
+		expand(table);
+	}
+
+	int hash = hashFunction(word) % table->buckets.size();
```
Ничто не мешает хеш-функции вернуть отрицательное число, и оператор `%` возвращает знак делимого, так что всё может упасть. И const бы не помешал

yurii-litvinov commented on Lab9/Task1/Homework/HashTable.cpp:
```diff
@@ -0,0 +1,263 @@
+
+bool exists(const string & word, HashTable const *table)
+{
+	int hash = hashFunction(word) % table->buckets.size();
```
И тут

nmamro committed:
Smooth changes

- Fixed type of hash function
  + changed types of "hash" var-s
- Changed method of filling HashTable
  - Now we pass created table to readFromFile(), so readFromFile() doesn't return anything
